src/ray/raylet/local_object_manager.cc

```diff
@@ -374,8 +377,14 @@ void LocalObjectManager::AsyncRestoreSpilledObject(
   // Restore the object.
   RAY_LOG(DEBUG) << "Restoring spilled object " << object_id << " from URL "
                  << object_url;
-  if (!node_id.IsNil()) {
-    RAY_CHECK(spilled_objects_url_.count(object_id) > 0);
+  if (is_external_storage_type_fs_ && spilled_objects_url_.count(object_id) == 0) {
+    RAY_CHECK(!node_id.IsNil());
+    // If the object wasn't spilled yet on this node, just return. The caller should retry
+    // in this case.
+    if (callback) {
```

Comment:
Can you log an INFO?

Reply:
I think this could be a little bit spammy since it will happen every 10ms now. What about DEBUG?